src/osgPlugins/gdal/ReaderWriterGDAL.cpp

```diff
@@ -557,14 +557,20 @@ class ReaderWriterGDAL : public osgDB::ReaderWriter
                     (unsigned char *)imageData,
                     osg::Image::USE_NEW_DELETE);
 
-                if (texCoordRange) image->setUserData(texCoordRange);
+                if (texCoordRange) 
+                {
+                    image->setUserData(texCoordRange);
+                    delete texCoordRange;
```

Comment:
This change introduces a dangling pointer and with it a bug, it's very much a step backwards.  As Kent suggests the OSG has reference counting that will ensure texCoordRange will be deleted automatically when the image gets deleted. 

What tool are you using to "detect" leaks?  It may well be that it's reporting false positives, or perhaps you are just interpreting things incorrectly as the changed code path doesn't have a leak.

As general tightening up of the code the appropriate thing to do is use a ref_ptr<<osgDB::ImageOptions::TexCoordRange> rather than C pointer.  I have also added a protected destructor for the TexCoordRange to prevent users for incorrectly attempting to delete a reference counted object (most other classes in the OSG already do this.)